packages/pn-personafisica-webapp/src/component/Contacts/SpecialContacts.tsx

```diff
@@ -155,44 +172,65 @@ const SpecialContacts = ({ recipientId, legalAddresses, courtesyAddresses }: Pro
     validateOnMount: true,
     validationSchema,
     onSubmit: (values) => {
-      initValidation(
-        values.addressType,
-        values.s_pec || values.s_mail || internationalPhonePrefix + values.s_phone,
-        recipientId,
-        values.sender,
-        async (status: 'validated' | 'cancelled') => {
-          if (status === 'validated') {
-            // reset form
-            formik.resetForm();
-            await formik.validateForm();
+      if (values.addressType) {
+        initValidation(
+          values.addressType,
+          values.s_pec || values.s_mail || internationalPhonePrefix + values.s_phone,
+          recipientId,
+          values.sender.id,
+          async (status: 'validated' | 'cancelled') => {
+            if (status === 'validated') {
+              // reset form
+              formik.resetForm();
+              await formik.validateForm();
+            }
           }
-        }
-      );
+        );
+      }
     },
   });
 
+  const handleFilterAutocomplete = (options: Array<Party>) => {
+    if (senderInputValue.length >= 4) {
+      return options.filter((item: Party) =>
+        String(item.name).toLowerCase().includes(senderInputValue.toLowerCase())
+      );
+    }
+    return options;
+  };
+
+  const renderOption = (props: any, option: Party) => (
+    <MenuItem {...props} value={option.id} key={option.id} role="option">
+      <DropDownPartyMenuItem name={option.name} />
+    </MenuItem>
+  );
+
+  const getOptionLabel = (option: Party) => option.name || '';
+
+  const handleChangeInput = (newInputValue: string) => {
+    setSenderInputValue(newInputValue);
+  };
+
   const handleChangeTouched = async (e: ChangeEvent) => {
     formik.handleChange(e);
     await formik.setFieldTouched(e.target.id, true, false);
   };
 
-  const senderChangeHandler = (e: ChangeEvent) => {
+  const senderChangeHandler = async (e: any, newValue: any) => {
```

Comment:
Avoid use of any type. The e should be ChangeEvent. Try to match newValue type as expected (Party?)

Reply:
I changed the types according to how the autocomplete function wanted it tell me if it's ok